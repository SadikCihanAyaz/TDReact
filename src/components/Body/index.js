import React, { useState } from "react";

const Body = (props) => {

const [count, setCount] = useState(0);

    const handleClick = () => {
        setCount(count + 1);
    }
    return(
        <>
            <h1 data-testid="title2">{props.title}</h1>
            <p data-testid="counter">{count}</p>
            <h1>TEST</h1>
            <button id="cihan2" type="button4" onClick={handleClick} data-testid="button5">BUTTON</button>
        </>

    );
}

export default Body;