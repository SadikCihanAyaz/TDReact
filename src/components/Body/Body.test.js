import React from 'react';
import Body from '../Body';
import {render} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { unmountComponentAtNode } from "react-dom";


describe("This suit is to test the Body component", ()=>{
    test('Snapshot of Body', () => {
        const {asFragment} = render(<Body title="hello world"/>);
        expect(asFragment()).toMatchSnapshot()
    });
    test("finding title", ()=>{
        const {getByText} = render(<Body title="Jay Rocks"/>);
        expect(getByText("Jay Rocks")).toBeInTheDocument("Jay Rocks")
    })
    test("finding title with test ID", () => {
        const {getByTestId} = render(<Body title="Jay Rocks"/>);
        expect(getByTestId("title2")).toHaveTextContent("Jay Rocks");
    })
    test("check Counter initial value", ()=>{
        const {getByTestId} = render(<Body title="Jay Rocks"/>);
        expect(getByTestId("counter")).toHaveTextContent("0")
    })
    test("check Counter value after a tap", ()=>{
        const {getByTestId} = render(<Body title="I Love Kavinie"/>);
        const button = getByTestId("button5");
        userEvent.click(button)
        expect(getByTestId("counter")).toHaveTextContent("1")
    })
    test("control new system", () => {
       
    })
})