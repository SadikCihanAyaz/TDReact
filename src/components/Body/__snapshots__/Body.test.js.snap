// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the Body component Snapshot of Body 1`] = `
<DocumentFragment>
  <h1
    data-testid="title2"
  >
    hello world
  </h1>
  <p
    data-testid="counter"
  >
    0
  </p>
  <button
    data-testid="button5"
    id="cihan2"
    type="button4"
  >
    BUTTON
  </button>
</DocumentFragment>
`;
